File: zad3/3.py
def suf_pref(str1, str2):
    for i in range(min(len(str1), len(str2)) + 1)[1:][::-1]:
        if str1[:i] == str2[-i:]:
            return i
    return 0

# Test cases
print(suf_pref("cde", "abcde"))  # Output: 3

print(suf_pref("xyz", "abcde"))  # Output: 0

print(suf_pref("abc", "abcde"))  # Output: 0

print(suf_pref("deghsaj", "abcde"))  # Output: 2

print(suf_pref("a", "abcde"))  # Output: 0

print(suf_pref("e", "abcde"))  # Output: 1

print(suf_pref("abcdef", "abcde"))  # Output: 5

print(suf_pref("bcdez", "abcde"))  # Output: 4

File: lab/zad3/3.c
#include <stdio.h>
#include <string.h>

int suf_pre(char* str1, char* str2) {
    int len1 = strlen(str1);
    int len2 = strlen(str2);
    int min_len = len1 < len2 ? len1 : len2;

    for (int i = min_len; i > 0; i--) {
        if (memcmp(str1, str2 + len2 - i, i) == 0) {
            return i;
        }
    }

    return 0;
}


int main() {
    printf("%d\n", suf_pre("cde", "abcde"));  // Output: 3
    printf("%d\n", suf_pre("xyz", "abcde"));  // Output: 0
    printf("%d\n", suf_pre("abc", "abcde"));  // Output: 0
    printf("%d\n", suf_pre("deghsaj", "abcde"));  // Output: 2
    printf("%d\n", suf_pre("a", "abcde"));  // Output: 0
    printf("%d\n", suf_pre("e", "abcde"));  // Output: 1
    printf("%d\n", suf_pre("abcdef", "abcde"));  // Output: 5
    printf("%d\n", suf_pre("bcdez", "abcde"));  // Output: 4
}
